File: 984. String Without AAA or BBB/withoutAAABBB.java
class Solution {
    public String strWithout3a3b(int a, int b) {
        int size = 3; 
 
        int total = a + b; 
        int len = size - 1;  
        
        String max, min; 
        int maxInt, minInt; 
        
        if(a > b){
            max = "a"; 
            min = "b";
            maxInt = a; 
            minInt = b; 
        }else{
            max = "b"; 
            min = "a"; 
            maxInt = b; 
            minInt = a; 
        }
        
        String output = ""; 
        
        while(total > 0){
            
            if(minInt == maxInt){
             len = 1; 
            }
            
            for(int i = 0; i < len; i++){
                if(maxInt > 0){
                    output += max; 
                    maxInt--; 
                    total--; 
                }
            }
            if(minInt > 0){
                output += min; 
                minInt--; 
                total--; 
            }
            
        }
        
        return output; 
    }
    
    
}

File: 984. String Without AAA or BBB/withoutAAABBB.rb
# @param {Integer} a
# @param {Integer} b
# @return {String}
def str_without3a3b(a, b)
    total = a + b
    len = 3 - 1; 
    
    max = String::new
    min = String::new
    output = String::new
    maxint = 0
    minint = 0
    
    if(a > b)
        max = "a"
        min = "b"
        maxint = a
        minint = b
    else
        max = "b"
        min = "a"
        maxint = b
        minint = a
    end 
        
    while total > 0
        if minint == maxint
            len = 1
        end 
        
        for i in 1..len
            if maxint > 0
                output << max
                maxint-= 1
                total -= 1
            end
        end
         if minint > 0
                output << min
                minint -= 1
                total -= 1
            end
        
    end 
    return output 
    
end
